pkg/server/plugin/nodeattestor/httpchallenge/httpchallenge.go

```diff
@@ -0,0 +1,272 @@
+	}
+
+	if req.CoreConfiguration == nil {
+		return nil, status.Error(codes.InvalidArgument, "core configuration is required")
+	}
+
+	if req.CoreConfiguration.TrustDomain == "" {
+		return nil, status.Error(codes.InvalidArgument, "trust_domain is required")
+	}
+
+	trustDomain, err := spiffeid.TrustDomainFromString(req.CoreConfiguration.TrustDomain)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "trust_domain is invalid: %v", err)
+	}
+
+	pathTemplate := httpchallenge.DefaultAgentPathTemplate
+	if len(hclConfig.AgentPathTemplate) > 0 {
+		pathTemplate, err = agentpathtemplate.Parse(hclConfig.AgentPathTemplate)
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "failed to parse agent svid template: %q", hclConfig.AgentPathTemplate)
+		}
+	}
+
+	var dnsPatterns []*regexp.Regexp
+	for _, r := range hclConfig.DNSPatterns {
+		re := regexp.MustCompile(r)
+		dnsPatterns = append(dnsPatterns, re)
+	}
+
+	agentNamePattern := regexp.MustCompile("^[a-zA-z]+[a-zA-Z0-9-]$")
+
+	allowNonRootPorts := true
+	if hclConfig.AllowNonRootPorts != nil {
+		allowNonRootPorts = *hclConfig.AllowNonRootPorts
+	}
+
+	tofu := true
+	if hclConfig.TOFU != nil {
+		tofu = *hclConfig.TOFU
+	}
+
+	mustUseTOFU := false
+	switch {
+	// User has explicitly asked for a required port that is untrusted
+	case hclConfig.RequiredPort != nil && *hclConfig.RequiredPort >= 1024:
+		mustUseTOFU = true
+	// User has just chosen the defaults, any port is allowed
+	case hclConfig.AllowNonRootPorts == nil && hclConfig.RequiredPort == nil:
+		mustUseTOFU = true
+	// User explicitly set AllowNonRootPorts to true and no required port specified
+	case hclConfig.AllowNonRootPorts != nil && *hclConfig.AllowNonRootPorts && hclConfig.RequiredPort == nil:
+		mustUseTOFU = true
+	}
+
+	if !tofu && mustUseTOFU {
+		return nil, status.Errorf(codes.InvalidArgument, "you can not turn off trust on first use (TOFU) when non root ports are allowed")
+	}
+
+	p.setConfiguration(&configuration{
+		trustDomain:       trustDomain,
+		pathTemplate:      pathTemplate,
+		dnsPatterns:       dnsPatterns,
+		requiredPort:      hclConfig.RequiredPort,
+		allowNonRootPorts: allowNonRootPorts,
+		agentNamePattern:  agentNamePattern,
+		tofu:              tofu,
+	})
+
+	return &configv1.ConfigureResponse{}, nil
+}
+
+// SetLogger sets this plugin's logger
+func (p *Plugin) SetLogger(log hclog.Logger) {
+	p.log = log
+}
+
+func (p *Plugin) getConfig() (*configuration, error) {
+	p.m.Lock()
+	defer p.m.Unlock()
+	if p.config == nil {
+		return nil, status.Errorf(codes.FailedPrecondition, "not configured")
+	}
+	return p.config, nil
+}
+
+func (p *Plugin) setConfiguration(config *configuration) {
+	p.m.Lock()
+	defer p.m.Unlock()
+	p.config = config
+}
+
+func buildSelectorValues(hostName string) []string {
+	var selectorValues []string
+
+	selectorValues = append(selectorValues, "httpchallenge:hostname:"+hostName)
```

Comment:
Selector values don't include the name of the plugin.
```suggestion
	selectorValues = append(selectorValues, "hostname:"+hostName)
```